test/storage/StorageTest.cc

```diff
@@ -0,0 +1,123 @@
+// Copyright 2017 Rice University, COMP 413 2017
+
+#include <easylogging++.h>
+#include <gtest/gtest.h>
+#include <zk_nn_client.h>
+#include "zkwrapper.h"
+#include "ClientNamenodeProtocolImpl.h"
+
+#include "StorageMetrics.h"
+
+#define ELPP_THREAD_SAFE
+
+INITIALIZE_EASYLOGGINGPP
+
+static const int NUM_DATANODES = 3;
+
+using asio::ip::tcp;
+using client_namenode_translator::ClientNamenodeTranslator;
+
+namespace {
+
+class StorageTest : public ::testing::Test {
+ protected:
+  virtual void SetUp() {
+    int error_code;
+    zk = std::make_shared<ZKWrapper>(
+        "localhost:2181,localhost:2182,localhost:2183", error_code, "/testing");
+    assert(error_code == 0);  // Z_OK
+
+    // Start the namenode in a way that gives us a local pointer to zkWrapper.
+    unsigned short port = 5351;
+    nncli = new zkclient::ZkNnClient(zk);
+    nncli->register_watches();
+    nn_translator = new ClientNamenodeTranslator(port, nncli);
+  }
+
+  // Objects declared here can be used by all tests below.
+  zkclient::ZkNnClient *nncli;
+  ClientNamenodeTranslator *nn_translator;
+  std::shared_ptr<ZKWrapper> zk;
+};
+
+/**
+ * The following is an example of using StorageMetrics.
+ * This test puts a file into RDFS, then prints 2 metrics (SD and % space used)
+ */
+TEST_F(StorageTest, testExample) {
+  asio::io_service io_service;
+
+  RPCServer namenodeServer = nn_translator->getRPCServer();
+  std::thread(&RPCServer::serve, namenodeServer, std::ref(io_service))
+      .detach();
+
+  ASSERT_EQ(0, system("python "
+                          "/home/vagrant/rdfs/test/integration/generate_file.py"
```

Comment:
Is there a way we could do a relative filepath here, it seems like this is brittle.

Reply:
i think we should keep it absolute. otherwise this call depends on the directory you run the test from